modules/base/rendering/screenspacetimevaryingimageonline.cpp

```diff
@@ -0,0 +1,234 @@
+        std::string jsonFilePath;
+    };
+#include "screenspacetimevaryingimageonline_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+ScreenSpaceTimeVaryingImageOnline::ScreenSpaceTimeVaryingImageOnline(
+                                                      const ghoul::Dictionary& dictionary)
+    : ScreenSpaceRenderable(dictionary)
+    , _jsonFilePath(JsonFileInfo, "")
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+
+    _jsonFilePath = absPath(p.jsonFilePath).string();
+    addProperty(_jsonFilePath);
+
+    _jsonFilePath.onChange([this]() {
+        loadJsonData(_jsonFilePath);
+        });
+}
+
+documentation::Documentation ScreenSpaceTimeVaryingImageOnline::Documentation() {
+    return codegen::doc<Parameters>("base_screenspace_time_varying_image_online");
+}
+
+bool ScreenSpaceTimeVaryingImageOnline::initialize() {
+    loadJsonData(_jsonFilePath.value());
+    return ScreenSpaceRenderable::initialize();
+}
+
+bool ScreenSpaceTimeVaryingImageOnline::deinitializeGL() {
+    _texture = nullptr;
+    return ScreenSpaceRenderable::deinitializeGL();
+}
+
+void ScreenSpaceTimeVaryingImageOnline::loadJsonData(const std::string& path) {
+    std::ifstream file(path);
+    if (!file.is_open()) {
+        throw ghoul::RuntimeError(std::format("Could not open JSON file at '{}'", path));
+        return;
+    }
+
+    nlohmann::json json;
+    file >> json;
+
+    if (json.find("files") == json.end()) {
+        throw ghoul::RuntimeError(std::format(
+            "Error loading JSON file. No 'files' was found in '{}'", path
+        ));
+    }
+
+    _timestamps.clear();
+    _urls.clear();
+
+    for (const auto& entry : json["files"]) {
+        const std::string& timestamp = entry["timestamp"].get<std::string>();
+        double j2000 = Time::convertTime(timestamp);
+        _timestamps.push_back(j2000);
+        _urls[j2000] = entry["url"].get<std::string>();
+    }
+
+    std::sort(_timestamps.begin(), _timestamps.end());
+    computeSequenceEndTime();
+}
+
+void ScreenSpaceTimeVaryingImageOnline::computeSequenceEndTime() {
+    if (_timestamps.size() > 1) {
+        double first = _timestamps.front();
+        double last = _timestamps.back();
+        double avg = (last - first) / static_cast<double>(_timestamps.size() - 1);
+        // Extend end time so the last value remains visible for one more interval
+        _sequenceEndTime = last + avg;
+    }
+}
+
+void ScreenSpaceTimeVaryingImageOnline::update() {
+    if (_timestamps.empty()) {
+        return;
+    }
+
+    double current = global::timeManager->time().j2000Seconds();
+
+    if (current < _timestamps.front() || current >= _sequenceEndTime) {
+        _activeIndex = -1;
+        _texture = nullptr;
+        _currentUrl.clear();
+        return;
+    }
+
+    if (current >= _timestamps.front() && current < _sequenceEndTime) {
+        int idx = activeIndex(current);
+        if (idx != _activeIndex) {
```

Comment:
```suggestion
        if (int idx = activeIndex(current);  idx != _activeIndex) {
```

Not necessary, but just food for thought